gammapy/modeling/sampler.py

```diff
@@ -0,0 +1,97 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Sampler parameter classes."""
+
+import logging
+import ultranest
+#from .sampler import SamplerLikelihood
+
+__all__ = ["Sampler", "SamplerResult","SamplerLikelihood"]
+
+
+
+
+class Sampler:
+    """Sampler class.
+
+    The sampler class provides a uniform interface to multiple sampler backends.
+    Currently available: "UltraNest", "emcee" in #TODO .
+
+    Parameters
+    ----------
+    backend : {"ultranest"}
+        Global backend used for sampler. Default is "ultranest".
+
+    #TODO : describe all parameters
+    """
+
+    def __init__(self, live_points=100, frac_remain=0.5, log_dir=None, resume=False, step_sampler=False):
+        self._sampler = None
+        self.min_num_live_points = live_points
+        self.frac_remain = frac_remain
+        self.log_dir = log_dir
+        self.step_sampler = step_sampler
+        self.resume = resume
+
+    @staticmethod
+    def _parse_datasets(datasets):
+        from gammapy.datasets import Dataset, Datasets
+
+        if isinstance(datasets, (list, Dataset)):
+            datasets = Datasets(datasets)
+#        return datasets, datasets.parameters.free_parameters
+        return datasets, datasets.parameters
```

Comment:
could remove @staticmethod and move to `gammapy.datasets.utils`  (and same in `Fit` class)

Reply:
I'm going to do that in a separate PR

Reply:
Done in #5646, and updated this PR.